sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -380,6 +380,11 @@ class SQLContext(@transient val sparkContext: SparkContext)
     jsonRDD(json.rdd, schema)
   }
 
+  @Experimental
+  def inferJsonSchema(json: String): StructType = {
```

Comment:
Instead of introducing a new interface, seems it will be better to extract general-purpose utility functions (https://issues.apache.org/jira/browse/SPARK-5260).